shared/src/commonMain/kotlin/com/myworkout/shared/app/usecases/LoginUseCase.kt

```diff
@@ -0,0 +1,28 @@
+package com.myworkout.shared.app.usecases
+
+import com.myworkout.shared.app.data.repositories.token.TokenRepository
+import com.myworkout.shared.app.data.services.AuthenticationService
+import com.myworkout.shared.domain.core.Response
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+
+class LoginUseCase(
+    private val authenticationService: AuthenticationService,
+    private val tokenRepository: TokenRepository
+) : UseCase<LoginUseCase.LoginUseCaseInput, Boolean> {
+
+    override suspend fun invoke(t: LoginUseCaseInput): Flow<Boolean> {
+        val response = authenticationService.authenticate(t.userName, t.password)
+
+        if (response is Response.Success) {
+            tokenRepository.token = response.result
```

Comment:
Might we need to capture the failure scenario here that  emits a `false`

Reply:
Not right here, but we are doing that in line 22 if the response is not a success.

Reply:
Ah, yea got it 👍